src/app/EventLogging.h

```diff
@@ -53,15 +55,55 @@ class EventLogger : public EventLoggingDelegate
  * context tags to be interpreted within the schema identified by
  * `ClusterID` and `EventId`.
  *
- * @param[in] apDelegate The EventLoggingDelegate to serialize the event data
+ * LogEvent has 3 variant, for aEventData that have method GetFabricIndex with return type FabricIndex or
+ * DataModel::Nullable<chip::FabricIndex>, it would use first two logEvent function twhich can process the fabric index, for
+ * aEventData that don't have GetFabricIndex, it would use tird LogEvent function.
+ * @param[in] aEventData  The event cluster object
  * @param[in] aEndpoint    The current cluster's Endpoint Id
  * @param[in] aUrgent    The EventOption Type, kUrgent or kNotUrgent
  * @param[out] aEventNumber The event Number if the event was written to the
  *                         log, 0 otherwise. The Event number is expected to monotonically increase.
  *
  * @return CHIP_ERROR  CHIP Error Code
  */
-template <typename T>
+template <typename T,
+          typename std::enable_if_t<std::is_class<T>::value && DataModel::IsFabricScoped<T>::value &&
+                                        std::is_same<decltype(std::declval<T>().GetFabricIndex()), FabricIndex>::value,
```

Comment:
What other sort of type would GetFabricIndex be returning?

Reply:
Two king of returned.
In acl cluster, for event with admin fabric index without nullable, it would have
FabricIndex GetFabricIndex() const { return adminFabricIndex; } where adminFabricIndex is not nullable type, its type is FabricIndex

In doorlock cluster, for event with fabric index with nullable, it would have
DataModel::Nullablechip::FabricIndex etFabricIndex() const { return fabricIndex; } where fabricindex is not nullable type.